### DISCUSSION TITLE
是否考虑插件共享？会不会有什么风险

### DISCUSSION THREAD

**Discussion by klarkxy** [howmanybots/onebot#90] [想法 / 建议]
最近在思考一个事情，很多人用bot其实都是用同样的几个功能。但是在安装的时候显然需要自己安装bot，然后装各种插件。
然后进行运行。
真正需要对程序代码修改的只是少部分人。
那么，可不可以做一个去中心化的集群/云，用户将自己的bot客户端连到云上，开发者将自己的服务注册到云上，用户通过网页或者私聊或者其他方式，选择加载、开关指定的服务。
不知道大家有没有什么比较好的想法，欢迎讨论。

**Comment by stdrc** [comment]
这大概就是一个支持 OneBot 标准的“图灵机器人”吧，不过去中心化的话，用户的 bot 总要连接到一个服务器，这个服务器怎么选呢

**Comment by klarkxy** [comment]
> 这大概就是一个支持 OneBot 标准的“图灵机器人”吧，不过去中心化的话，用户的 bot 总要连接到一个服务器，这个服务器怎么选呢

目前我自己的想法是开一个仓库用来存愿意分享自己服务器作为节点的IP。但是由此会带来一些安全上的问题，会给人带去麻烦。
不知道有没有更合适的思路。

**Comment by ExerciseBook** [comment]
其实我觉得这个确实不错，如果做出来了我能贡献一个 GitHub 机器人（bushi）


感觉可以配合 go-cqhttp 之类的支持多连接反向 ws 的机器人来使用。

**Comment by undefined-moe** [comment]
总结：有人捐钱就能做

**Comment by super1207** [comment]
可不可以通过插件的形式提供功能呢，插件通过ws之类的方式连上onebot之后，就可以公开(注册)插件里面的某项功能，然后其它插件就能够使用公开的功能。这样就不需要中心服务器了，没人捐钱也能做:)

**Comment by stdrc** [comment]
> 可不可以通过插件的形式提供功能呢，插件通过ws之类的方式连上onebot之后，就可以公开(注册)插件里面的某项功能，然后其它插件就能够使用公开的功能。这样就不需要中心服务器了，没人捐钱也能做:)

可能 @klarkxy 的意图就是为了避免“装插件”这一步吧，只需要运行比如 go-cqhttp 就能用，而不需要自己折腾 nonebot 或是其他 bot 框架

**Comment by crazywhalecc** [comment]
这个其实可以任意一个人开放一个服务，但是前提是得有精力维护一个框架和功能贡献，而且前提是框架的生态比较健壮有人持续输出插件创意。现在这点貌似只有nonebot可能还会好一点。说白了就是一个机器人的实际功能的运算平台？

**Comment by crazywhalecc** [comment]
不过感觉应该很有意思，接入一个公共的机器人服务，先开一个比较全的问答插件，自己可设置问答。然后自定义功能和一些自定义的 API 请求类的可以通过 Issue 或 PR 等提交。被信任方可以选择将框架和算法开源（前提是要有开源精神）。。。

**Comment by ExerciseBook** [comment]
@crazywhalecc 做一个问答功能出来也是不错的

**Comment by stdrc** [comment]
> 说白了就是一个机器人的实际功能的运算平台？

Bot as a Service 哈哈
